lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1637,6 +1638,16 @@ class HSAQueue final : public KalmarQueue
             DBOUT(DB_CMD2, "No future found in wait, enqueued marker into " << *this << "\n");
         }
         back()->getFuture()->wait();
+        if (HCC_FLUSH_ON_WAIT) {
+            // aggressively cleanup resources
+            // but keep back() as a valid HSAOp, so decrement current insert index twice
```

Comment:
it's unclear to me the reason for keep the recent HSAOp, could you elaborate?

Reply:
During testing, I was getting seg faults if I didn't leave the last op.  I think I figured out why as I review the code.  `isEmpty()` is using `back()` without checking it for nullptr.

Reply:
I think I'll merge in this PR as-is and address the isEmpty() issue in another PR.